examples/citi_bike/online_lp/citi_bike_lp.py

```diff
@@ -0,0 +1,187 @@
+                        self._transfer[tick][station][neighbor_idx] for neighbor_idx in range(self._num_neighbor)
+                    )
+                ), f"TotalTransferFrom_T{tick}_S{station}"
+                problem += (
+                    self._transfer_to[tick][station] == lpSum(
+                        self._transfer[tick][neighbor][self._station_neighbor_list[neighbor].index(station)]
+                        for neighbor in range(self._num_station)
+                        if station in self._station_neighbor_list[neighbor][:self._num_neighbor]
+                    )
+                ), f"TotalTransferTo_T{tick}_S{station}"
+
+        for tick in range(1, self._num_tick):
+            for station in range(self._num_station):
+                problem += (
+                    self._inventory[tick][station] == (
+                        self._inventory[tick - 1][station]
+                        + supply[tick - 1, station] - self._fulfillment[tick - 1][station]
+                        + self._transfer_to[tick - 1][station] - self._transfer_from[tick - 1][station]
+                    )
+                ), f"Inventory_T{tick}_S{station}"
+                problem += (
+                    self._safety_inventory[tick][station] <= (
+                        self._inventory[tick - 1][station]
+                        + supply[tick - 1, station] - self._fulfillment[tick - 1][station]
+                        - self._transfer_from[tick - 1][station]
+                    )
+                ), f"SafetyInventory_T{tick}_S{station}"
+
+    def _set_objective(self, problem: LpProblem):
+        fulfillment_gain = lpSum(
+            math.pow(self._fulfillment_time_decay_factor, tick) * lpSum(
+                self._fulfillment[tick][station] for station in range(self._num_station)
+            ) for tick in range(self._num_tick)
+        )
+
+        safety_inventory_reward = self._safety_inventory_reward_factor * lpSum(
+            self._safety_inventory[tick][station] for station in range(self._num_station)
+            for tick in range(self._num_tick)
+        )
+
+        transfer_cost = self._transfer_cost_factor * lpSum(
+            self._transfer_to[tick][station] for station in range(self._num_station)
+            for tick in range(self._num_tick)
+        )
+
+        problem += (fulfillment_gain + safety_inventory_reward - transfer_cost)
+
+    def _formulate_and_solve(
+        self, env_tick: int, init_inventory: np.ndarray, demand: np.ndarray, supply: np.ndarray
+    ):
+        problem = LpProblem(
+            name = f"Citi_Bike_Repositioning_from_tick_{env_tick}",
+            sense=LpMaximize,
+        )
+        self._init_variables(init_inventory=init_inventory)
+        self._add_constraints(problem=problem, demand=demand, supply=supply)
+        self._set_objective(problem=problem)
+        problem.solve(GLPK(msg=0))
+        # print(f"Problem {problem.name}: {LpStatus[problem.status]}")
+
+    def get_transfer_list(
```

Comment:
mv public interface up.

Reply:
Public functions in the earlier part of these class, private functions in later part?